go/service/chat_boxer.go

```diff
@@ -169,3 +187,31 @@ func (b *chatBoxer) verify(data []byte, si chat1.SignatureInfo, prefix libkb.Sig
 	_, err := sigInfo.Verify()
 	return (err == nil)
 }
+
+// validSenderKey checks that the key is active for sender at ctime.
+func (b *chatBoxer) validSenderKey(sender gregor1.UID, key []byte, ctime gregor1.Time) (bool, error) {
+	kbSender, err := keybase1.UIDFromString(hex.EncodeToString(sender.Bytes()))
```

Comment:
I agree, the above is ugly, but that's how it is done elsewhere...I'll add the converter.